File: programmers/Level 1/폰켓몬.py
def solution(nums) :
    answer = []
    for i in nums :
        if i not in answer :
            answer.append(i)
            if len(answer) == len(nums) // 2 :
                break
            
    return len(answer)

"""
min을 활용한 다른 사람 풀이
def solution(ls):
    return min(len(ls)/2, len(set(ls)))
"""

File: programmers/코딩테스트 고득점 Kit/해시/폰켓몬.java
import java.util.*;

class Solution{
    public int solution(int[] nums){
        int max_P = nums.length / 2;
        
        HashSet<Integer> set = new HashSet<>();
        for(int n : nums){
            set.add(n);
        }
        if (max_P > set.size()){
            return set.size();
        }
        else{
            return max_P;
        }
    }
}
